refactor(tools): use typed context keys for kubeconfig and work_dir

Vinay-Khanagavi opened the pull request:
- Replaced string keys with a custom contextKey type for kubeconfig and work_dir
- Prevents context key collisions and follows Go best practices
- Resolves static analysis/linter warnings
- No functional changes

Closes #174

mikebz commented:
Looks like there are more occurrences of these strings, have you considered making this a more global change?

```
mcp.go: ctx = context.WithValue(ctx, "work_dir", s.workDir)
pkg/tools/bash_tool.go: workDir := ctx.Value("work_dir").(string)
pkg/tools/trivy_tool.go:        workDir := ctx.Value("work_dir").(string)
pkg/tools/tools.go:     ctx = context.WithValue(ctx, "work_dir", opt.WorkDir)
pkg/tools/kubectl_tool.go:      workDir := ctx.Value("work_dir").(string)
```

Vinay-Khanagavi commented:
Thanks for pointing that out!
now it's global change across the codebase, please review:

- Replaced all string-based context keys for "work_dir" and "kubeconfig" with the exported, typed constants WorkDirKey and KubeconfigKey.
- Updated all relevant files (tools.go, bash_tool.go, kubectl_tool.go, trivy_tool.go) to use these constants.
- This prevents key collisions, follows Go best practices, and resolves static analysis warnings.

Let me know if you spot any other places or have further suggestions!  

All string-based context key usages for "kubeconfig" and "work_dir" have been refactored to use typed constants throughout the tools package. 

I noticed there’s also a usage in **mcp.go** (`ctx = context.WithValue(ctx, "work_dir", s.workDir)`). Since that file isn’t part of the tools package, I didn’t update it in this PR. Please let me know if you’d like me to update mcp.go as well, or if there’s any context I should be aware of for that file!

janetkuo reviewed (commented):
`mcp.go` has the same problem as well. Would you fix that too?

Vinay-Khanagavi commented:
Done @janetkuo!
I’ve updated mcp.go to use the typed context keys as well.
Branch feat/context-key-type is ready to merge. Thanks so much for your support!
